### ISSUE TITLE
Earlystopping 'auto' mode is misleading and errorprone

### ISSUE THREAD

**Issue by qmarcou** [keras-team/keras#18798]
This is basically a duplicate/transfer (with delay) of tensorflow/tensorflow#59036, with slight adjustments now that more cases are handled by the "auto" mode.
 
 ### Issue Type

Bug

### Keras Version

2.14.0, commit id on master https://github.com/keras-team/keras/commit/1f0a013b02bbafc8a016353a0e60ba94895fea07

### Current Behaviour?

Trying to use the EarlyStopping callback, the documentation states for the 'auto' mode:
https://github.com/keras-team/keras/blob/1f0a013b02bbafc8a016353a0e60ba94895fea07/keras/callbacks/early_stopping.py#L32-L36
> mode: One of `{"auto", "min", "max"}`. In `min` mode,
        training will stop when the quantity
        monitored has stopped decreasing; in `"max"`
        mode it will stop when the quantity
        monitored has stopped increasing; in `"auto"`
        mode, the direction is automatically inferred
        from the name of the monitored quantity.

In fact the corresponding keras code is just:
https://github.com/keras-team/keras/blob/1f0a013b02bbafc8a016353a0e60ba94895fea07/keras/callbacks/early_stopping.py#L99-L111

This means that for any metric that is basically not the accuracy or an auc the 'auto' mode will default to 'min', while a vast majority of performance metrics are designed to be increasing functions of performance (recall, precision, F score,  NDCG, etc).

This documentation is extremely misleading (I have wasted a lot of computation time myself due to this), and given this behavior I think we can safely say that almost nothing is inferred from the monitored metric.

In order to protect other users from this trap, I would advise to remove the 'auto' mode and its associated argument altogether.

Another solution, if removing the default argument is too much of a breaking change, would be to include at least all the metrics from base keras and return an error when the the provided metric is not in base keras. This would however have a higher maintenance cost, since a new entry should be added every time a new metric is introduced into keras. A test going through all possible metrics in keras would reduce the risk of errors. That being said, because the current implementation relies on the `name` attribute of the metric, and this name is arbitrary this solution remains error prone without a complete rewrite of the checking mechanism (e.g a user using the "auroc" name would not be properly handled, since it does not `endswith` 'auc')

Depending on the preferred solution I can provide a PR to fix this behavior. As user having been trapped by this default argument, I really do think fixing this is important, and I'm sure other have been trapped without noticing...

Thanks for all the good work!

### Standalone code to reproduce the issue

```shell
EarlyStopping(monitor='val_precision',
               mode='auto')
# this callback will tend to minimize precision...
```

**Comment by SuryanarayanaY** [comment]
Hi @qmarcou ,

Thanks for reporting. We also have an internal ticket on same which is yet to review. 

Similar issues found #[TF-Forum](https://discuss.tensorflow.org/t/confusion-in-earlystopping-mode-argument/14080) and #[SO](https://stackoverflow.com/questions/69744405/tensorflow-earlystopping-stops-too-early).

**Comment by fchollet** [comment]
I pushed an extensive fix. `EarlyStopping` will either recognize a known metrics, or will raise an error asking for the mode to be set explicitly. Check that it works for you (at HEAD).
